app/src/main/java/com/example/voyagerx/repository/database/Converters.kt

```diff
@@ -0,0 +1,47 @@
+package com.example.voyagerx.repository.database
+
+import androidx.room.TypeConverter
+import com.example.voyagerx.repository.model.Launch
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+
+class Converters {
+
+    @TypeConverter
+    fun fromStringtoListOfLaunch(value: String?): List<Launch>?{
+        if(value == null){
+            return null
+        }
+        val gson = Gson()
+        val listType = object: TypeToken<List<Launch>>(){}.type
+        return gson.fromJson(value, listType)
+    }
+
+    @TypeConverter
+    fun fromListOfLaunchToString(list: List<Launch>?): String?{
+        if(list == null){
+            return null
+        }
+        val gson = Gson()
+        return gson.toJson(list)
+    }
+
+    @TypeConverter
+    fun fromStringToListOfString(value: String?): List<String>?{
+        if(value == null){
+            return null
+        }
+        val gson = Gson()
+        val listType = object: TypeToken<List<String>>(){}.type
+        return gson.fromJson(value, listType)
+    }
+
+    @TypeConverter
+    fun fromListOfStringToString(list: List<String>?): String?{
+        if(list == null){
+            return null
+        }
+        val gson = Gson()
+        return gson.toJson(list)
+    }
+}
```

Comment:
What's the purpose of these conversions? I'm not too familiar with Android Room

Reply:
I updated them, so some of these converters are not the same, but typeconverters are used to convert complicated data types to simple data types so that they can be stored in the db (and retrieved as the type that they were originally).

Reply:
great question and answer!! This is what code review is about!!! 🥳

Reply:
Awesome!! Thanks Vivian 😄